app/src/main/java/fi/project/petcare/ui/screens/PetProfileFormScreen.kt

```diff
@@ -1,20 +1,254 @@
+import androidx.compose.material.icons.filled.Category
+import androidx.compose.material.icons.filled.MonitorWeight
+import androidx.compose.material.icons.filled.Numbers
+import androidx.compose.material.icons.filled.Pets
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 
 @Composable
-fun PetProfileFormScreen(onSubmit: (String, Int, String) -> Unit) {
+fun PetProfileFormScreen() {
+    // State variables to hold form values and validation errors
+    var name by remember { mutableStateOf("") }
+    var weight by remember { mutableStateOf("") }
+    var species by remember { mutableStateOf("") }
+    var breed by remember { mutableStateOf("") }
+    var age_months by remember { mutableStateOf("") }
+    val genderOptions = listOf("Male", "Female", "Other")
+    var petNameError by remember { mutableStateOf("") }
+    var genderError by remember { mutableStateOf("") }
+    var weightError by remember { mutableStateOf("") }
+    var speciesError by remember { mutableStateOf("") }
+    var breedError by remember { mutableStateOf("") }
+    var age_monthsError by remember { mutableStateOf("") }
+    var microchipId by remember { mutableStateOf("") }
+    var selectedGender by remember { mutableStateOf("") }
+    var microchipIdError by remember { mutableStateOf("") }
     Column(
-        modifier = Modifier.fillMaxSize().padding(16.dp),
-        verticalArrangement = Arrangement.Center,
+        modifier = Modifier
+            .fillMaxHeight()
+            .padding(16.dp),
         horizontalAlignment = Alignment.CenterHorizontally
     ) {
-        Text(text = "Pet Profile Form")
-        // You can place the PetProfileForm composable here
-    }
+        Text(
+            text = "Create a account for you pet",
+            style = MaterialTheme.typography.headlineMedium,
+            modifier = Modifier.padding(bottom = 8.dp)
+
+
+        )
+
+        // Microchip ID field with validation
+        OutlinedTextField(
+            value = microchipId,
+            onValueChange = {
+                microchipId = it
+                microchipIdError = if (it.isEmpty()) "Microchip ID cannot be empty" else ""
+            },
+            keyboardOptions = KeyboardOptions(
+                keyboardType = KeyboardType.Text,
```

Comment:
Since microchipId is of type Int, consider using Numeric keyboard type here and for weight field as well.